src/server/models/cacheable_queries/lib.js

```diff
@@ -0,0 +1,16 @@
+
+
+export const modelWithExtraProps = (obj, Model, props) => {
+  const newObj = { ...obj }
+  const extraProps = {}
+  props.forEach(prop => {
```

Comment:
Do we need this? Seems like we're exactly coping `newObj` into `extraProps` unless I'm missing something. Couldn't `newObj` be used on line 12 instead of `extraProps`?

Reply:
I'll add some documentation on the function -- note that we delete properties from newObj, so in the spirit of 'functional' style, where you generally want to avoid altering objects inputted into functions.  Thus we do the copy.  This is an odd dance for the use-case of attaching properties to Models but allowing Model validation to take place.